### PR TITLE
Enable/disable preview for all the datasets when publishing Kedro-Viz from CLI

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves [1870](https://github.com/kedro-org/kedro-viz/issues/1870) 

## Development notes

* `--include-preview` flag added for Viz CLI:
* By default, kedro viz deploy and kedro viz build will have preview disabled for all the datasets
* Introduced a ClassVar `is_all_previews_enabled` for `DataNodeMetadata` class which controls the enable/disable functionality for all the datasets

## QA notes

* Run `kedro viz build --include-preview` to enable preview for all datasets
* Run `kedro viz deploy --include-preview` to enable preview for all datasets in the deployed version

**NOTE:** As discussed with @rashidakanchwala and @stephkaiser , `kedro viz run` will not have a preview flag and preview will be set to True by default. If the users want to change this behavior, they need to do this from UI. Designs will be updated soon (cc: @stephkaiser )

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by stephkaiser — APPROVED** [review]

**Comment by rashidakanchwala** [comment]
There's a ticket here #1705  , we should add this to next sprint

File: package/kedro_viz/api/rest/responses.py [hunk 1]
```diff
@@ -452,7 +457,7 @@ def save_api_pipeline_response_to_fs(pipelines_path: str, remote_fs: Any):
             raise exc
 
 
-def save_api_responses_to_fs(path: str, remote_fs: Any):
+def save_api_responses_to_fs(path: str, remote_fs: Any, are_datasets_previewable: bool):
```
Comment (merelcht) [inline package/kedro_viz/api/rest/responses.py:460]: This is personal preference, but I would change `are_datasets_previewable` to `preview_datasets`.

To me `are_datasets_previewable` sounds like it's checking whether it's possible to preview these datasets or not, where when reading the code it should mean whether preview has been enabled or not. 

In my proposed change `preview_datasets=True` means preview is enabled and `preview_datasets=False` means it's disabled.

File: package/kedro_viz/models/flowchart.py [hunk 2]
```diff
@@ -806,7 +812,11 @@ def set_run_command(cls, _):
     @field_validator("preview")
     @classmethod
     def set_preview(cls, _):
-        if cls.data_node.is_preview_disabled() or not hasattr(cls.dataset, "preview"):
+        if (
+            cls.data_node.is_preview_disabled()
+            or not hasattr(cls.dataset, "preview")
+            or not cls.are_datasets_previewable
```
Comment (merelcht) [inline package/kedro_viz/models/flowchart.py:818]: What's the difference here between `cls.data_node.is_preview_disabled()` and `cls.are_datasets_previewable`? Semantically, it sounds very similar, so I think we should think carefully about how these flags are named so it's easy to understand what's happening for future team members. 

Also in terms of functionality, what does it mean for preview to be disabled on a node level vs class level?

**Review by merelcht — COMMENTED** [review]
The code looks good, but I have some reservations about the naming of the new `are_datasets_previewable` boolean. It's getting confusing with the `is_preview_disabled` functionality we have.

**Comment by merelcht** [comment]
One thought is perhaps to name the new boolean after the new `--include-preview` flag, so it's clear those two are connected?

File: package/kedro_viz/models/flowchart.py (hunk 2 above)
Comment (rashidakanchwala) [reply package/kedro_viz/models/flowchart.py:818]: Agree with @merelcht. 

i am also thinking for clarity we could possibly have :- 
- is_preview_disabled on node level (and explain this in the docstring, that this disabled via node level happens when the user specifies 'preview=false' in the DataCatalog)
- is_all_previews_disabled on class level (and once again explain in docstring this is to disable previews for all nodes and can be done via CLI and UI)

File: package/kedro_viz/api/rest/responses.py (hunk 1 above)
Comment (ravi-kumar-pilla) [reply package/kedro_viz/api/rest/responses.py:460]: okay, I too had a hard time renaming this :) .... so I want to make it differentiable to the `is_preview_disabled` which corresponds to each instance of DataNodeMetadata class, where-as the plural form `are_datasets_previewable` is a class variable. I personally prefer having is/are before a boolean variable/method to be clear of the return/data type.

File: package/kedro_viz/models/flowchart.py (hunk 2 above)
Comment (jitu5) [reply package/kedro_viz/models/flowchart.py:818]: @merelcht As discussed with @rashidakanchwala the existing `is_preview_disabled` will rename to `is_preview_enabled` and new boolean variable `are_datasets_previewable` to `is_all_previews_enabled` .

**Review by rashidakanchwala — APPROVED** [review]
LGTM. Can we please also create a a ticket/PR to change  `is_preview_disabled` to `is_preview_enabled` so it is consistent. 

thanks both @jitu5 and @ravi-kumar-pilla :D

**Comment by jitu5** [comment]
@rashidakanchwala Sure I will create a separate PR for change of `is_preview_disabled` to `is_preview_enabled`

**Review by ravi-kumar-pilla — APPROVED** [review]
LGTM. Thank you

**Comment by jitu5** [comment]
@stephkaiser Requesting you approval on disable preview from CLI, few days back, I demoed it you. please let me know if you have any questions.

**Comment by stephkaiser** [comment]
> @stephkaiser Requesting you approval on disable preview from CLI, few days back, I demoed it you. please let me know if you have any questions.

no questions from me! looks good, thanks Jitendra!

**Review by stephkaiser — APPROVED** [review]

**Pull request merged**